plugins/amqp_compressed_proof_plugin/amqp_compressed_proof_plugin.cpp

```diff
@@ -219,8 +219,11 @@ void compressed_proof_generator::on_applied_transaction(const chain::transaction
 void compressed_proof_generator::on_accepted_block(const chain::block_state_ptr& bsp) {
    std::set<action_entry> action_entries_this_block;
 
-   if(my->onblock_trace)
-      action_entries_this_block.emplace((*my->onblock_trace)->action_traces.front());
+   // verify block_num of onblock_trace because last block could have been aborted & new block have no onblock
+   if(my->onblock_trace && (*my->onblock_trace)->block_num == bsp->block_num) {
```

Comment:
This looks correct to me, but it's tricky.  If the previous block is the same, then onblock will consistently either succeed or fail, since it's always starting from the same state.  If the previous block is different, then it must be an ancestor of the previous block corresponding to the onblock trace, because pushing any block will reset onblock_trace.

Reply:
block_header timestamp is a slot and everything else should be consistent as well... hmm, ok, nice thanks!

Reply:
No.  I didn't consider the timestamp.  The block timestamp can skip time and is observable from within a contract.

Reply:
There's also the case of activating `preactive_feature`, which happens before `onblock`.

Reply:
Hmm, back to the drawing board then.